src/main/java/hudson/plugins/promoted_builds/integrations/jobdsl/PromotionContext.java

```diff
@@ -1,108 +1,185 @@
 package hudson.plugins.promoted_builds.integrations.jobdsl;
 
 import groovy.lang.Closure;
+import groovy.lang.MetaClass;
 import groovy.util.Node;
+import groovy.util.NodeBuilder;
+import groovy.util.NodeList;
+import groovy.util.XmlParser;
 import hudson.plugins.promoted_builds.PromotionCondition;
 import hudson.plugins.promoted_builds.conditions.DownstreamPassCondition;
 import hudson.plugins.promoted_builds.conditions.ParameterizedSelfPromotionCondition;
 import hudson.plugins.promoted_builds.conditions.SelfPromotionCondition;
 import hudson.plugins.promoted_builds.conditions.UpstreamPromotionCondition;
+import javaposse.jobdsl.dsl.DslContext;
+import javaposse.jobdsl.dsl.Item;
+import javaposse.jobdsl.dsl.JobManagement;
+import javaposse.jobdsl.dsl.helpers.step.StepContext;
+import javaposse.jobdsl.plugin.DslEnvironment;
+import org.codehaus.groovy.runtime.InvokerHelper;
+import org.codehaus.groovy.runtime.MethodClosure;
 
 import java.util.ArrayList;
 import java.util.List;
 
-import javaposse.jobdsl.dsl.Context;
-import javaposse.jobdsl.dsl.helpers.step.StepContext;
+import static javaposse.jobdsl.plugin.ContextExtensionPoint.executeInContext;
 
-import javaposse.jobdsl.plugin.DslEnvironment;
+class PromotionContext extends Item {
```

Comment:
Is the compatibility issue a concern with data persistence as performed by Jenkins or related to actual API usage by third party developers?
For the former, the class is not persisted so it should be an issue.
For the later, a new class could be introduced (but then the existing class will be unused).